app/src/common/shared/com/igalia/wolvic/utils/SystemUtils.java

```diff
@@ -35,24 +32,12 @@ public class SystemUtils {
     public static final long TWO_DAYS_MILLIS = 172800000;
     public static final long ONE_WEEK_MILLIS = 604800000;
 
-    public static final void restart(@NonNull Context context) {
-        Intent i = new Intent(context, VRBrowserActivity.class);
-        i.setPackage(BuildConfig.APPLICATION_ID);
-        context.startActivity(i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK));
-    }
-
-    public static final void scheduleRestart(@NonNull Context context, long delay) {
-        Intent i = new Intent(context, VRBrowserActivity.class);
-        i.setPackage(BuildConfig.APPLICATION_ID);
-        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-
-        PendingIntent mPendingIntent = PendingIntent.getActivity(context, 0, i,
-                PendingIntent.FLAG_CANCEL_CURRENT | PendingIntent.FLAG_IMMUTABLE);
-        AlarmManager mgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
-        mgr.set(AlarmManager.RTC, System.currentTimeMillis() + delay, mPendingIntent);
-
-        System.exit(0);
+    public static void restart(@NonNull Context context) {
+        PackageManager pm = context.getPackageManager();
+        Intent intent = pm.getLaunchIntentForPackage(context.getPackageName());
+        Intent mainIntent = Intent.makeRestartActivityTask(intent.getComponent());
+        context.startActivity(mainIntent);
```

Comment:
You can even use a single line for these two statements

Reply:
Done!